nvflare/fuel/message/message_bus.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+
+
+class MessageBus:
+    _instance = None
+    _lock = threading.Lock()
+
+    def __new__(cls):
+        with cls._lock:
+            if not cls._instance:
+                cls._instance = super(MessageBus, cls).__new__(cls)
+                cls._instance.subscribers = {}
+                cls._instance.message_store = {}
+        return cls._instance
+
+    def subscribe(self, topic, callback):
+        if topic not in self.subscribers:
+            self.subscribers[topic] = []
+        self.subscribers[topic].append(callback)
+
+    def publish(self, topic, message):
+        if topic in self.subscribers:
+            for callback in self.subscribers[topic]:
+                callback(message)
```

Comment:
The callback should also take "topic" as an input arg. This allows the same cb to be used for multiple topics.